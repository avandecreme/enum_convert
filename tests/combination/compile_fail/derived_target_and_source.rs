use enum_convert::{EnumFrom, EnumInto};

#[derive(EnumInto)]
#[enum_into(Target)]
enum Source {
    Unit,
    Tuple(i32, &'static str),
    Struct { x: i32, y: i32 },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from]
    Unit,
    #[enum_from]
    Tuple(i64, String),
    #[enum_from]
    Struct { x: i64, y: i64 },
}

fn main() {}
