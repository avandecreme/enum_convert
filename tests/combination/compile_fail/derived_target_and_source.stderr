error[E0119]: conflicting implementations of trait `From<Source>` for type `Target`
  --> tests/combination/compile_fail/derived_target_and_source.rs:12:10
   |
4  | #[derive(EnumInto)]
   |          -------- first implementation here
...
12 | #[derive(EnumFrom)]
   |          ^^^^^^^^ conflicting implementation for `Target`
   |
   = note: this error originates in the derive macro `EnumFrom` (in Nightly builds, run with -Z macro-backtrace for more info)
